docs/tutorials/core-from-scratch.ipynb

```diff
@@ -0,0 +1,369 @@
+    "plt.axhline(0, color=\"k\")\n",
+    "plt.xlabel(\"eccentric anomaly\")\n",
+    "plt.ylabel(r\"error from Kepler solver\");"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9",
+   "metadata": {},
+   "source": [
+    "We enabled double precision above (although that is by no means a requirement for this code), and we can see here that the results are correct to within $<10^{-15}$ of absolute error, which should do just fine for any practical applications.\n",
+    "This error does increase for large eccentricities, but even then it should always perform better than $\\sim 10^{-12}$, and a full analysis is beyond the scope of this tutorial.\n",
+    "\n",
+    "It's also worth testing the performance of this implementation.\n",
+    "Since we execute these tutorials on Read the Docs, the specific results here will depend on the actual allocated hardware, but here's how you would benchmark this operation on your system:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "10",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Using a large number of test values to minimize effects of Python overhead\n",
+    "mean_anom_bench = jax.device_put(\n",
+    "    jnp.linspace(0, 2 * jnp.pi, 500_000)[:-1]\n",
+    ").block_until_ready()\n",
+    "kepler_solver_impl(mean_anom_bench, ecc).block_until_ready()\n",
+    "%timeit kepler_solver_impl(mean_anom_bench, ecc).block_until_ready()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11",
+   "metadata": {},
+   "source": [
+    "On my 2020 M1 MacBook Pro, I get a runtime of `7 ms` per loop, and in each loop we're solving Kepler's equation 500,000 times, so that means that each solve costs about `14 ns`.\n",
+    "For comparison, on the same system, the Kepler solver from [RadVel](https://github.com/California-Planet-Search/radvel/blob/b279a88710b628f70c9de4045f86e2f864674dc5/src/_kepler.pyx#L24) which is implemented in C/Cython takes `35 ms` per loop.\n",
+    "\n",
+    "We should note that the point here is not to pick on RedVel, and let's be clear that this is by no means a complete comparison!\n",
+    "Instead, we want to highlight that in some cases JIT-compiled JAX code can get similar performance to that provided by optimized low level code with Python bindings.\n",
+    "\n",
+    "Another benefit of this pure-JAX implementation is that it can be natively executed on hardware accelerators like a GPU or TPU.\n",
+    "For comparison, on an Nvidia V100 GPU, the above benchmark takes about `0.1 ms` (nearly 2 orders of magnitude faster than my CPU).\n",
+    "\n",
+    "### Supporting differentiation\n",
+    "\n",
+    "Another key feature of JAX (besides the fact that its JIT compiler produces very fast code) is that it supports automatic differentiation.\n",
+    "While it is already possible to differentiate the above code, that wouldn't produce the most efficient method.\n",
```

Comment:
Maybe very briefly discuss why the built-in AD wouldn't be efficient? 
Even someone short like "...since tracing through the `kepler_solver_impl` to obtain the derivative is inefficient when we can explicitly write the derivative given below."

Reply:
Added!